python/ray/serve/_private/deployment_state.py

```diff
@@ -2223,26 +2223,18 @@ def _choose_pending_migration_replicas_to_stop(
 
         # Stop replicas whose deadline is up
         for replica in replicas:
+            assert replica.actor_node_id in deadlines
+
             curr_timestamp_ms = time.time() * 1000
             timeout_ms = replica._actor.graceful_shutdown_timeout_s * 1000
-            if (
-                replica.actor_node_id in deadlines
-                and curr_timestamp_ms >= deadlines[replica.actor_node_id] - timeout_ms
-            ):
+            if curr_timestamp_ms >= deadlines[replica.actor_node_id] - timeout_ms:
                 to_stop.append(replica)
             else:
                 remaining.append(replica)
 
         # Stop excess PENDING_MIGRATION replicas when new "replacement"
         # replicas have transitioned to RUNNING. The replicas with the
         # earliest deadlines should be chosen greedily.
-        def order(deadline: int):
-            if deadline:
-                return deadline
-            else:
-                return float("inf")
-
-        # remaining.sort(key=lambda r: order(deadlines[r.actor_node_id]))
         remaining.sort(key=lambda r: deadlines[r.actor_node_id])
```

Comment:
Is there any chance that different version replicas are running at the same time? If so, different replicas may have different `graceful_shutdown_timeout_s` values, and we should adjust the deadlines by those when sorting.

Reply:
Good catch, I think that's possible. This is a small edge case though because outdated replicas should also get killed in `check_and_update_replicas`. I will create a follow up PR to address this so that this PR is not blocked on this (this is not newly introduced in this PR).

Reply:
Sounds good.